src/Element/Select2.php

```diff
@@ -202,10 +203,23 @@ public static function preRenderSelect($element) {
     $current_language = \Drupal::languageManager()->getCurrentLanguage();
     $current_theme = \Drupal::theme()->getActiveTheme()->getName();
     $select2_theme_exists = \Drupal::service('library.discovery')->getLibraryByName($current_theme, 'select2.theme');
+
+    // Placeholder should be taken from #placeholder property if it set.
+    // Otherwise we can take it from '#empty_option' property.
+    if (isset($element['#placeholder'])) {
+      $placeholder = $element['#placeholder'];
+    }
+    elseif (isset($element['#empty_option'])) {
+      $placeholder = $element['#empty_option'];
+    }
+    else {
+      $placeholder = $required ? new TranslatableMarkup('- Select -') : new TranslatableMarkup('- None -');
+    }
```

Comment:
It might also possible here, but I am not sure we should do that
```suggestion
    $placeholder = $element['#placeholder'] ?? $element['#empty_option'] ?? $required ? new TranslatableMarkup('- Select -') : new TranslatableMarkup('- None -');
```

Reply:
I this it would be better to format this code like this (for better understanding):
```
$placeholder = $element['#placeholder'] 
  ?? $element['#empty_option'] 
  ?? ($required ? new TranslatableMarkup('- Select -') : new TranslatableMarkup('- None -'));
```
split it to several lines and added braces to last part